files/external_node_v2.rb

```diff
@@ -342,22 +345,24 @@ def watch_and_send_facts(parallel)
       upload_facts_parallel(http_fact_requests)
     else
       certname = ARGV[0] || raise("Must provide certname as an argument")
-      # send facts to Foreman - enable 'facts' setting to activate
-      # if you use this option below, make sure that you don't send facts to foreman via the rake task or push facts alternatives.
-      #
-      if SETTINGS[:facts]
-        req = generate_fact_request certname, "#{puppetdir}/yaml/facts/#{certname}.yaml"
-        upload_facts(certname, req)
-      end
+
       #
       # query External node
       begin
         result = ""
         Timeout.timeout(tsecs) do
```

Comment:
I'm concerned about putting the fact upload into this timeout block without increasing the default timeout, as Foreman responds to fact uploads very slowly. I'd possibly prefer leaving it outside - between `begin` and `Timeout.timeout`?